### PR TITLE
Insert method with tests added

### LINKED ISSUE #35: No insert method
Would be nice if we could use the insert method like this one: https://msdn.microsoft.com/pl-pl/library/sey5k5z4(v=vs.110).aspx

### PR DESCRIPTION
**Pull request by mstrzoda**
Fixes #35

### CODE REVIEW

**Comment by mstrzoda** [comment]
hi guys, this is my first pull request on github, i hope i didn't break anything ;)

File: dist/linq.d.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@
  * Copyright © 2016 Flavio Corpa. All rights reserved.
  *
  */
-export class List<T> {
+export declare class List<T> {
```
Comment (kutyel) [inline dist/linq.d.ts]: Why the changes to the definition file? What did you use to generate this?

File: dist/linq.js [hunk 2]
```diff
@@ -17,9 +18,7 @@ var List = (function () {
      * Defaults the elements of the list
      */
     function List(elements) {
-        elements = elements || [];
-        // we clone the array to prevent side effects
-        elements = elements.slice(0);
+        if (elements === void 0) { elements = []; }
```
Comment (kutyel) [inline dist/linq.js]: I agree to this changes but since this JavaScript file is a transpiled version, what option in the `.tsconfig` file did you add to change the output? 😄

File: linq.ts [hunk 3]
```diff
@@ -171,6 +171,17 @@ export class List<T> {
...
 
     /**
+     * Inserts an element into the List<T> at the specified index.
+     */
+    public Insert(index: number, element: T): void | TypeError {
+        if (index < 0 || index > this._elements.length) {
+            throw new TypeError('Index is out of range.');
+        }
+
+        this._elements.splice(index, 0, element);
```
Comment (kutyel) [inline linq.ts]: As almost all the methods in this LINQ implementation follow a functional programming approach, could you return a `new List()` with the new element inserted instead of mutating the previous one? 😉

File: test/list.ts [hunk 4]
```diff
@@ -80,6 +80,62 @@ test('Add', t => {
     t.is(list.First(), 'hey');
 });
 
+test('Insert', t => {
```
Comment (kutyel) [inline test/list.ts]: All the tests look great! If you notice I have just *one* iteration per method, so could you please put all these tests inside the same block? They are just assertions of the same method. 😉

File: linq.ts (hunk 3 above)
Comment (mstrzoda) [inline linq.ts]: This method is very similar to Add method, that returns void as well, so why i should return list here? Also the definition on MSDN does not contain returning anything from Insert method: https://msdn.microsoft.com/en-us/library/sey5k5z4(v=vs.110).aspx

Comment (kutyel) [inline linq.ts]: Ok, you are right, the closer we adhere to the msdn, the better 😉  Just change the tests and we are good to go! 🎉

**Comment by kutyel** [comment]
Thanks very much for your hard work! 👏 👍
